pkg/artifactory/resource_artifactory_pull_replication.go

```diff
@@ -4,12 +4,73 @@ import (
 	"context"
 	"encoding/json"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-
 	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
+var pullReplicationSchema = map[string]*schema.Schema{
+	"url": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		RequiredWith:     []string{"username", "password"},
+		ValidateDiagFunc: validation.ToDiagFunc(validation.IsURLWithHTTPorHTTPS),
+		Description:      "(Optional) URL for local repository replication. Required for local repository, but not needed for remote repository.",
+	},
+	"socket_timeout_millis": {
+		Type:             schema.TypeInt,
+		Optional:         true,
+		Computed:         true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+	},
+	"username": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		RequiredWith:     []string{"url", "password"},
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+		Description:      "(Optional) Username for local repository replication. Required for local repository, but not needed for remote repository.",
+	},
+	"password": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Sensitive:        true,
+		RequiredWith:     []string{"url", "username"},
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+		Description:      "(Optional) Password for local repository replication. Required for local repository, but not needed for remote repository.",
+	},
+	"enabled": {
+		Type:     schema.TypeBool,
+		Optional: true,
+		Computed: true,
+	},
+	"sync_deletes": {
+		Type:     schema.TypeBool,
+		Optional: true,
+		Computed: true,
+	},
+	"sync_properties": {
+		Type:     schema.TypeBool,
+		Optional: true,
+		Computed: true,
```

Comment:
Do need Computed: true for attributes like sync_deletes, sync_properties and sync_statistics?

Reply:
Ideally no, they shouldn't have `Computed` but I'm not changing them for this PR.